### PR TITLE
ADD closes #30

### LINKED ISSUE #30: fix failing `struct` case
currently, this test case fails: https://github.com/janestreet/bin_prot/blob/master/test/non_integers_repr.ml#L747 

the struct and test are defined in `tests/struct.rs`:
```
#[derive(Serialize, Deserialize, Debug, PartialEq)]
struct C {
    y: CInner,
    z: Option<i64>,
}

#[derive(Serialize, Deserialize, Debug, PartialEq)]
struct CInner {}
```

```
    let c0 = C {
        y: CInner {},
        z: None,
    };

    let mut c0_expected = vec![0x00, 0x00, 0x01, 0x01];
    c0_expected.reverse();
```

the test output is:
```
thread 'test_serialize_structs' panicked at 'assertion failed: `(left == right)`
  left: `[1, 1, 0, 0]`, // expected
 right: `[0]`', tests/struct.rs:100:13 // actual
```

would be good to double check that the rust struct is the same as the OCaml struct, and if so, then this case needs to be fixed.

### PR DESCRIPTION
**Pull request by mmacedoeu**
The dummy CInner(pub ()) is able to be serialized as 0x00 0x00 since there is no inline record in rust I have to strip last 2 bytes for now

### CODE REVIEW

**Review by willemolding — CHANGES_REQUESTED** [review]
I'm not sure if this is quite the right fix. I added some comments to the original issue with some links which hopefully clear things up https://github.com/ChainSafe/serde-bin-prot/issues/30#issuecomment-891440705

**Comment by mmacedoeu** [comment]
I have updated the test including the extra bytes and added new comment on original issue explaining the reasoning behind multiple choices.

File: tests/non_integers_repr.rs [hunk 1]
```diff
@@ -144,12 +144,16 @@ fn test_record2() {
...
-        .. .. .. .. .. .. .. .. .. .. .. .. 0x00 0x00 0x00 -> B {
-            y: BInner { w: 0, x: 0 },
+        .. .. .. .. .. .. .. .. .. .. .. .. 0x00 0x00 0x00 0x00 0x01-> B3 {
+            y: BInner2::V1(BInner4::V0(BInner { w: 0, x: 0 })),
             z: None,
         },
-        0x00 0x7f 0xff 0xff 0xff 0xfd 0x7f 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xfc -> B {
-            y: BInner { w: 9223372036854775807, x: 2147483647 },
+        0x00 0x7f 0xff 0xff 0xff 0xfd 0x7f 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xfc 0x00 0x01 -> B2 {
+            y: BInner3 { dummy: 1 , inner: BInner2::V0(BInner { w: 9223372036854775807, x: 2147483647 })},
```
Comment (noot) [inline tests/non_integers_repr.rs:152]: why are these two types different? in the bin_prot tests it looks like they're the same type, but maybe I'm missing something https://github.com/janestreet/bin_prot/blob/6c45b4249fc2c3a19574c4c788e2d296cae7fc30/test/non_integers_repr.ml#L745

**Review by noot — COMMENTED** [review]
looks good! just a question about the types used

File: tests/non_integers_repr.rs (hunk 1 above)
Comment (mmacedoeu) [reply tests/non_integers_repr.rs:152]: On ocaml original test the type is the same, on our rust side we can represent the same type in more than one way, I explored this using two different rust types in order to showcase this. When using this in a real application you could choose what works best case by case

**Review by willemolding — APPROVED** [review]

**Review by creativcoder — APPROVED** [review]

**Pull request merged**